### ISSUE TITLE
Scalar multiplication on 1

### ISSUE THREAD

**Issue by dmitriyb** [stargrave/gogost#6]
Hello!
I have checked your curve multiplication function and found that if degree is equal to 1 than we would get error with a bad degree value. I have checked the [RFC 7091](https://tools.ietf.org/html/rfc7091) and found that d and k restrictions are 0<d<q and 0<k<q. So d and k may be equal to 1. But in that case Exp() function in curve.go will return error "Bad degree value".

```go
func (c *Curve) Exp(degree, xS, yS *big.Int) (*big.Int, *big.Int, error) {
	dg := big.NewInt(0).Sub(degree, bigInt1)
	tx := big.NewInt(0).Set(xS)
	ty := big.NewInt(0).Set(yS)
	cx := big.NewInt(0).Set(xS)
	cy := big.NewInt(0).Set(yS)
	if dg.Cmp(zero) == 0 {
		return nil, nil, errors.New("Bad degree value")
	}
	for dg.Cmp(zero) != 0 {
		if dg.Bit(0) == 1 {
			c.add(tx, ty, cx, cy)
		}
		dg.Rsh(dg, 1)
		c.add(cx, cy, cx, cy)
	}
	return tx, ty, nil
}
```
Am I missing something here? Why we strictly check degree to be not equal to 1?
Thank you!

**Comment by stargrave** [comment]
You are right: that was a mistake. Actually I was going to compare with zero value. Thanks! Fixed in b7c8ff70c07784ae7d8d854f02c628e88c32a8a2 commit.

**Comment by dmitriyb** [comment]
Thanks! That works well!
